dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/BatchingDelayResolver.kt

```diff
@@ -0,0 +1,19 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal
+
+import com.datadog.android.api.storage.FeatureStorageConfiguration
+
+internal class BatchingDelayResolver {
```

Comment:
imo it is a bit too much of abstractions, let's maybe just mock values returned by the `wrappedFeature` in the `SdkFeature`?

Reply:
I get your point, in the same time I prefer to have them separated and do not mix responsibilities. It's way easier for unit testing. I also suspect that this logic will change later so in that case `SdkFeature` will not change anymore and we will just change the resolver. What do you think ?

Reply:
not a fan of the overhead introduced by the resolvers either, it would be cleaner to not use them IMO

Reply:
I will try to add these as private methods instead, just need to find a better way to test it without too much noise.